### ISSUE TITLE
[Enhancement] [General] Introduce Options for Override / Safeguard To Pre-Emptively Open Stash Overlay (i.e. Early Set Turn-In)

### ISSUE THREAD

**Issue by LeonChadwick** [ChaosRecipeEnhancer/ChaosRecipeEnhancer#504]
"not at full set threshold" reduce your threshold and save settings and press stash button again...

don't be silly, do what I ask when I ask, not give nonsense instructions like above.  If I want to trade in early let me!

**Comment by Mwoua** [comment]
Agreed, its annoying
(but thanks for the good work anyway)

**Comment by zsmith009** [comment]
Agreed. That used to be working for as long as I've known this tool, now this seems to have changed. Also, is there still a shortcut to remove the overlay?

**Comment by jeremydaniels999** [comment]
Bring back the ability to trade in sets early please.  Very annoying to have to go change settings in order to be able to click Stash again...

**Comment by AlexRFD** [comment]
Thanks for all the work @HiMarioLopez 

I haven't worked with C# before but it seems the only part controlling this is in these lines [link](https://github.com/ChaosRecipeEnhancer/EnhancePoEApp/blob/main/src/App/ChaosRecipeEnhancer.UI/Windows/SetTrackerOverlayViewModel.cs#L198-L207)

If that seems correct I can open a new PR with the required changes.

**Comment by HiMarioLopez** [comment]
I think the intent was there to help alleviate a lot of confusion by users who would click the button and see no items or sets highlighted. I clearly missed the mark with this change - for that, I apologize.

I will introduce an option to bypass this behavior in the next full release.

Thank you for your feedback and I appreciate your help with this.

**Comment by HiMarioLopez** [comment]
I have added the setting to enable early turn ins, and it will be enabled by default (similar to how it was before).

I had to make a few modifications to the set composition algorithms to respect this new setting, but it seems to be stable now. Related commit: https://github.com/ChaosRecipeEnhancer/EnhancePoEApp/commit/2159a96a3f9008d172da017c941983a249f0bb2f

Look out for an upcoming release in a few days. Thanks!
